File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        // int left = 0;
        // int right = numbers.length-1;
        // while(left<right){
        //     if(numbers[left]+numbers[right]<target){
        //         left++;
        //     }else if(numbers[left]+numbers[right]>target){
        //         right--;
        //     }else{
        //         return new int[]{left+1, right+1};
        //     }
        // }
        // return new int[]{0}; 

        int left = 0;
        int right = numbers.length -1;
        while(left<right){
            int sum = numbers[left]+numbers[right];
            if(sum<target){
                left++;
            }else if(sum>target){
                right--;
            }else{
                return new int[]{left+1,right+1};
            }
        }
        return new int[]{0};

    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        right = 0
        left = len(numbers)-1
        while right < left:
            sum = numbers[right]+numbers[left]
            if sum > target:
                left-=1
            elif sum < target:
                right+=1
            else:
                return [right+1,left+1]
        return -1


        